### PR TITLE
[DEV-5903] Finish up the POST API documentation

### PR DESCRIPTION
**Pull request by sejanick**
# Description
Finish up porting over the POST API documentation: https://developer.snapengage.com/post-api/post-api/

I combined the JSON and XML examples, since they should be the same information formatted differently.  That isn't ~exactly right, but I think the different examples with the same explanation makes sense here together rather than apart.

### Related Issue(s)
- https://snapengage.atlassian.net/browse/DEV-5903

# Ready-to-Merge Checklist
- [ ] Adds or updates dependencies? No.
- [x] Have you built these changes locally? Yes.

### CODE REVIEW

File: source/includes/_post-api-details.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
+### POST API Details
+
+This page will go over all of the possible fields that may be included within a POST of a support request. If a field does not have data associated with it, then it will not show up within the data.  The data may be returned in JSON or XML.
+
+**Retries**: The target system needs to respond with and HTTP 200 to confirm that the event has been received and processed. In the absence of a successful HTTP response, SnapEngage will retry the HTTP POST automatically for approximately 2 hours before sending it through email and flagging the event as sent.
+
+**Using PHP**: The content is a stream in JSON or XML format. You can not access this stream in the $`_`* variables of PHP. Please use this instead: *$json = fopen("php://input","r");*
```
Comment (samanthagmccormick) [inline source/includes/_post-api-details.md]: I think we try to put all code snippets in the code panel, so the little chunk of php code would probably be best in the php tab. It just sort of draws more attention to the code to have it on the right side, with the human description on the left. On the left side you could say something like "The content is a stream in JSON or XML format. You can not access this stream in the $`_`* variables of PHP. Please use the PHP sample code to the right, instead."

Comment (samanthagmccormick) [reply source/includes/_post-api-details.md]: Sort of like these - http://127.0.0.1:4567/?javascript#chatmessagesent-event

File: source/post-api/index.html.md [hunk 2]
```diff
@@ -22,6 +19,4 @@ Using the API, SnapEngage can automatically POST events to an external URL when
 
 An authentication key is not needed to use the POST API.
 
-### How to activate the POST API
-
-In the SnapEngage Admin Dashboard, under the "Integrations" tab, select "Open API" and enter the URL of where you want to receive the POST message when a new offline request or live chat is processed by SnapEngage. Once you are done making changes, just make sure to click the "Save" button.
\ No newline at end of file
+In the SnapEngage Admin Dashboard, under the "Integrations" tab, select "Open API" and enter the URL of where you want to receive the POST message when a new offline request or live chat is processed by SnapEngage. Once you are done making changes, just make sure to click the "Save" button.
```
Comment (samanthagmccormick) [inline source/post-api/index.html.md]: I think the screenshot is really necessary here - the one on the old site is old, so maybe you could take a new one, and temporarily host it on screencast or something. There is a way to host images on Github Pages sites but we should add a Jira to cross that bridge later. I think for now the important thing is to not forget to add the screenshots so just having a placeholder is fine.

File: source/includes/_post-api-details.md (hunk 1 above)
Comment (sejanick) [reply source/includes/_post-api-details.md]: I actually had changed this on purpose, as I didn't like how that looked locally for this particular example!  Can we look at this together in person to discuss this?

File: source/post-api/index.html.md (hunk 2 above)
Comment (sejanick) [reply source/post-api/index.html.md]: Good call!  I will add that in.

**Comment by samanthagmccormick** [comment]
It makes sense to combine JSON and XML into one page, but we are actually trying to keep the same exact URL structure so that we don't lose any SEO value so I would like to just keep the same URL structure as the old site. 

I also think now it is a little buried / not as easy to see that we have both JSON and XML return data formats for this API. For instance if I'm in support it would be easy for me to find a link for a client / potential customer here: 
image once they tell me they need XML format.

But now it's less obvious:
image

The client now has to dig to find this info.

**Comment by sejanick** [comment]
@samanthagmccormick I think I addressed all feedback as discussed, so this should be ready for another look when you get the chance.  Thanks!

**Comment by samanthagmccormick** [comment]
Thank you for making those changes! bitmoji

**Pull request merged**